setup.py

```diff
@@ -17,7 +17,8 @@
     long_description=long_description,
     long_description_content_type="text/markdown",
     license="Apache License Version 2.0",
-    packages=setuptools.find_packages(
+    packages=setuptools.find_namespace_packages(
```

Comment:
Wait, this probably will not work. The original `find_packages` finds packages using `__init__` file. I am wondering whether this change will omit our core source folder `forte`.

Reply:
I tried reinstalling `forte` with the updated `setup.py` file and checked the site-packages that got installed.
I could see forte root source folder in addition to ft and ftx (Not sure if this is what you meant though)

<img width="455" alt="Screen Shot 2022-02-19 at 9 58 12 PM" src="https://user-images.githubusercontent.com/14886942/154830602-f9dbff06-676c-4bd5-b0fb-81ac538d8e40.png">

Also, wouldn't our build pipeline on git throw an error too for most of our import statements in the source code, if it were to skip forte's root source folder?

Reply:
The CI pipeline fails for other reasons (due to our recent updates on Texar-pytorch), so it did fail so I don't know what happen to the full test.

I am not familiar with the behavior of `find_namespace_packages` when both types of packages exist so I have to be careful here. Would be great if we could find supporting documentation for this.

Reply:
By updating the master the pipeline seems to work, this is good news but let's be a little cautious and confirm the actual behavior of `find_namespace_packages`.

Reply:
[This](https://discuss.python.org/t/mixing-namespace-and-non-namespace-packages-bad-practice/7536/11) thread discusses a similar scenario I think. It says packaging works if we name and situate our packages correctly, but some devs suggest its not a best practice. Couldn't find much intricate documentation on namespace packaging, it's mostly vague. Let me know what you think.

Reply:
The thread you refer to is a different scenario, which is a scenario that we should avoid. 

In that thread, there is a package named `myproject` with an `__init__.py`, but he also wants to use `myproject. datagen` as a namespace package. This is not advised because if you install two projects using different methods, it will not work.

In our case, we plan to have `ftx` in multiple places, and we don't want to have an `__init__.py` at `ftx` level.  So this is a standard namespace packaging setup. This will work correctly.  We also have another package `forte`, which won't conflict with `ftx` anyways.

My question is that: what's the behavior of `find_namespace_packages`? Our previous approach is `find_packages` and we know that it will try to find all packages by looking for `__init__` files. We want to know whether `find_namespace_packages` will do the same thing and then in addition to that, find the namespace packages passed through the argument. In other words, is `find_namespace_packages` doing `find_packages`'s job and some additional work.

Reply:
Regarding that, setup tools documentation says `find_namespace_packages` is backwards compatible to `find_packages`. Hence, that should imply it does all that `find_packages` as a utility does and more. 

[Reference](https://setuptools.pypa.io/en/latest/setuptools.html)

Reply:
thanks for the reference, I didn't find it myself. it should be good now.